src/htmx.js

```diff
@@ -4910,23 +4910,28 @@ var htmx = (function() {
    * @returns {HtmxExtension[]}
    */
   function getExtensions(elt, extensionsToReturn, extensionsToIgnore) {
-    if (extensionsToReturn == undefined) {
+    const isParentScan = !!extensionsToReturn
+    if (!extensionsToReturn) {
       extensionsToReturn = []
     }
-    if (elt == undefined) {
+    if (!elt) {
       return extensionsToReturn
     }
-    if (extensionsToIgnore == undefined) {
+    if (!extensionsToIgnore) {
       extensionsToIgnore = []
     }
     const extensionsForElement = getAttributeValue(elt, 'hx-ext')
     if (extensionsForElement) {
       forEach(extensionsForElement.split(','), function(extensionName) {
-        extensionName = extensionName.replace(/ /g, '')
-        if (extensionName.slice(0, 7) == 'ignore:') {
+        extensionName = extensionName.trim()
+        if (extensionName.indexOf('ignore:') === 0) {
           extensionsToIgnore.push(extensionName.slice(7))
           return
         }
+        if (extensionName.indexOf('this-only:') === 0) {
+          if (isParentScan) return
```

Comment:
small thing but I don't like if statements w/o a {} (I'm hypocritical about this, but hey it's a code review)